swatch-contracts/src/test/java/com/redhat/swatch/contract/service/ContractServiceTest.java

```diff
@@ -263,10 +261,7 @@ void createPartnerContract_NotDuplicateContractThenPersist() {
     productTags.data(List.of("MH123"));
     when(syncResource.getSkuProductTags(any())).thenReturn(productTags);
 
-    Map<String, Object> stringObjectMap =
-        Map.of("subscriptionNumber", contract.getRedHatSubscriptionNumber());
-    when(contractRepository.getContract(stringObjectMap, true))
-        .thenReturn(Optional.of(existingContract));
+    when(contractRepository.getContracts(any())).thenReturn(List.of(existingContract));
```

Comment:
This is for partner contract where we need to get only single contract as explained above. Would like to see its equivalent if we are doing based on Specification.